types/src/shared/memoize.ts

```diff
@@ -1,29 +1,21 @@
 // Memoize function with TTL.
 // Usage:
-// const memoizedFn = memoize(fn, (fnArg1, fnArg2, ...) => `${fnArg1}-${fnArg2}`, 60 * 10 * 1000);
-// Cache expires after ttlMs since last usage of the memoized function.
+// const memoizedFn = memoize(fn, (fnArg1, fnArg2, ...) => `${fnArg1}-${fnArg2}`, 10);
+// The keys are evicted in LRU order after the cache reaches maxCapacity.
 export function memoize<T extends (...args: any[]) => any>(
   fn: T,
   resolver: (...args: Parameters<T>) => string,
-  ttlMs: number
+  maxCapacity: number
 ): T {
   const cache = new Map<string, ReturnType<T>>();
-  let ttlTimeout: NodeJS.Timeout | undefined = undefined;
 
-  const extendTTL = () => {
-    clearTimeout(ttlTimeout);
-    ttlTimeout = setTimeout(() => {
-      cache.clear();
-    }, ttlMs);
-  };
-  extendTTL();
   return function (...args: Parameters<T>): ReturnType<T> {
-    extendTTL();
-
     const key = resolver(...args);
```

Comment:
If I understand correctly, different memoize fns can collide / use the same keys, right ?